### DISCUSSION TITLE
RenovateBot maven scanning updates itself

### DISCUSSION THREAD

**Discussion by gaukharaya** [renovatebot/renovate#7966] [Request Help]
Hello, 

We are running renovate to scan our maven dependencies, in java based repos. All maven dependencies live in artifactory, at the end of the repo CI build a jar snapshot of the repo code itself is published to artifactory. 

Here is where the problem is: 

Our versioning setup mandates that we have a property called `revision` in root pom.xml which defines the major and minor versions, whereas the patch version is defined using the number of commits and is controlled by Cloudbees pipeline. This has the unfortunate downside of making renovatebot think that there is a later version available for itself, so it tries to update like this:

`0.1 -> 0.1.159`

I want to use `ignoreDeps`, but I am not sure how to tell renovate configuration to not update itself during any given repo scan?

**Comment by rarkins** [comment]
Can you clarify what you mean by "making renovatebot think that there is a later version available for itself"? Renovate isn't published as a Maven dependency so I don't understand based on the description so far.

And do you mean that you would want Renovate to update from `0.1` to `0.2` but just not from `0.1` to `0.1.159`? You might be able to achieve that by disabling patch updates for that package name, for example

**Reply by gaukharaya** [reply]
Hello thank you for replying.

service-a has major and minor version defined in root `pom.xml` (e.g. `1.0`), renovate treats it as specification of version of dependency to use not as a definition. It also doesn't help that service-a gets published to artifactory using the major and minor defined in `pom.xml` + a generated patch(e.g. `1.0.1`) 


 I think the solution here is to use repository config `renovate.json` in service A to:
```
{
  "extends": ["config:base"],
  "ignoreDeps": ["com.example:service-a"]
}
```

However I found that repository config despite being detected is not respected. Is there a reason why repository config overrides would be ignored? Could it be because our `"onboarding": false`?

```
DEBUG: Found renovate.json config file (repository=project/service-a)

DEBUG: Repository config (repository=project/service-a)

       "fileName": "renovate.json",

       "config": {"extends": ["config:base"], "ignoreDeps": ["com.example:service-a"]}
DEBUG: migrateAndValidate() (repository=project/service-a)

DEBUG: No config migration necessary (repository=project/service-a)

DEBUG: massaged config (repository=project/service-a)

       "config": {"extends": ["config:base"], "ignoreDeps": ["com.example:service-a"]}
DEBUG: migrated config (repository=project/service-a)

       "config": {"extends": ["config:base"], "ignoreDeps": ["com.example:service-a"]}
DEBUG: Found renovate.json config file (repository=project/service-a)

DEBUG: Repository config (repository=project/service-a)

       "fileName": "renovate.json",

       "config": {"extends": ["config:base"], "ignoreDeps": ["com.example:service-a"]}
DEBUG: migrateAndValidate() (repository=project/service-a)

DEBUG: No config migration necessary (repository=project/service-a)

DEBUG: massaged config (repository=project/service-a)

       "config": {"extends": ["config:base"], "ignoreDeps": ["com.example:service-a"]}
DEBUG: migrated config (repository=project/service-a)

       "config": {"extends": ["config:base"], "ignoreDeps": ["com.example:service-a"]}

```

**Reply by rarkins** [reply]
You've shown that the config is detected but not that it's ignored - the logs finish too soon.
